testing/test_collection.py

```diff
@@ -2031,3 +2031,36 @@ def test_namespace_packages(pytester: Pytester, import_mode: str):
             "      <Function test_module3>",
         ]
     )
+
+
+@pytest.mark.parametrize(
+    "parametrize_args, expected_indexs",
+    [
+        ("[(1, 1), (1, 1)]", "{'1-1': [[]0, 1[]]}"),
+        ("[(1, 1), (1, 2), (1, 1)]", "{'1-1': [[]0, 2[]]}"),
+        ("[(1, 1), (2, 2), (1, 1)]", "{'1-1': [[]0, 2[]]}"),
+        ("[(1, 1), (2, 2), (1, 2), (2, 1), (1, 1)]", "{'1-1': [[]0, 4[]]}"),
```

Comment:
I find the argument names and `expected_indexs` hard to read; how about we define a helper function:

```python
def fnmatch_escape_repr(obj):
    return re.sub(r"[*?[\]]", (lambda m: "[{}]".format(m.group())), repr(obj))
```